lib/rubocop/cop/commissioner.rb

```diff
@@ -64,6 +63,11 @@ def reset_errors
       end
 
       # TODO: Bad design.
+      def remove_irrelevant_cops(filename)
+        @cops.select! { |cop| cop.relevant_file?(filename) }
```

Comment:
You can also add `Cop#irrelevant_file?`. :-) (or `excluded_file?`, which is more in line with our terminology so far).